syntax = "proto3";

package notification;

service NotificationService {
  rpc SendNotification(NotificationRequest) returns (NotificationResponse);
  rpc GetNotifications(TokenRequest) returns (NotificationsResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message TokenRequest {
  string token = 1;
}

message NotificationRequest {
  string token = 1;
  string recipient = 2;
  string notification_type = 3;
  string subject = 4;
  string content = 5;
  map<string, string> notification_metadata = 6;
}

message NotificationResponse {
  int32 id = 1;
  string recipient = 2;
  string notification_type = 3;
  string subject = 4;
  string content = 5;
  map<string, string> notification_metadata = 6;
  string status = 7;
  string created_at = 8;
  string updated_at = 9;
}

message NotificationsResponse {
  repeated NotificationResponse notifications = 1;
}

message HealthCheckRequest {
}

message HealthCheckResponse {
  string status = 1;
  string service = 2;
} 